DOCS-3867: TrueCommand 2.2 & SCALE Clustering Documentation (#1424)
truenas/documentation: content/Solutions/Integrations/SMBClustering.md

 - /truecommand/clustering/manageclusters/
 - /truecommand/clustering/mountingclustervolumes/
---

{{< toc >}}

{{< include file="/content/_includes/ClusterIntro.md" type="page" >}}

## Warnings and Restrictions

{{< include file="/content/_includes/ClusterWarnings.md" type="page" >}}

## Requirements

{{< include file="/content/_includes/ClusterRestrictions.md" type="page" >}}

## Setting up the Environment

Configuring the cluster feature is a multi-step process that spans multiple systems.

{{< expand "Setup Guide (click to expand)" "v" >}}
{{< include file="/content/_includes/ClusterSetup.md" type="page" >}}
{{< /expand >}}

## Creating the Cluster

When the SCALE, AD, and TrueCommand environments are ready, log in to TrueCommand to cluster the SCALE systems.

1. Click the <span class="iconify" data-icon="mdi:server-network"></span> **Clusters** icon in the upper left. Click **NEW CLUSTER** to see the cluster creation options.

   ![CreateClusterSystemsNetwork](/images/TrueCommand/2.2/CreateClusterSystemsNetwork.png "Network Options for Clustered Systems")

2. Enter a unique name for the cluster and open the dropdown to select the systems to include in the cluster.

> 3. When each SCALE system is listed, open the **Network Address** dropdown for each system and choose the static IP address from the previously configured subnet dedicated for cluster traffic.

Reviewer: where are these system listed? Maybe add where they see the listed systems.

{{< hint warning >}}
Skipping this step is not recommended because there are no opportunities to reset the configuration after it is completed.
To go back and add the AD and SMB connection details requires deleting and remaking the cluster.
{{< /hint >}}

1. For each SCALE system, choose the IP address related to the primary subnet.
   This is typically the IP address used to connect the SCALE system to TrueCommand.
   Click **NEXT**.
2. Enter the Microsoft Active Directory credentials and click **NEXT**.
3. Verify the connection details are correct and click **SUBMIT**.

{{< hint info >}}
Creating a cluster has no visible effect on each SCALE web interface.
To verify the cluster is created and active, open the SCALE **Shell** and enter `gluster peer status`.
The command returns the list of SCALE IP addresses and current connection status.
{{< /hint >}}

## Creating Cluster Volumes

1. In the TrueCommand **Clusters** screen, find the cluster to use and click **CREATE VOLUME**.
   
   ![ClustersCreateVolumeDetails](/images/TrueCommand/2.2/ClustersCreateVolumeDetails.png "Add Cluster Volume: Details")
   
2. Enter a unique name for the cluster and select a **Type**.

>    ### Volume Types

Reviewer: Does this render in the toc? if not, just use the bold emphasis.

>    ### Volume Types

Author: ```suggestion
  **Volume Types**
```

   {{< expand "Current Types (click to expand)" "v" >}}
   {{< include file="/content/_includes/ClusterTypes.md" type="page" >}}
   {{< /expand >}}

3. After configuring the **Type**, enter a **Brick Size** based on the available storage from the clustered pools and your storage requirements.
4. Review the **Pools** for each SCALE system in the cluster and ensure that the desired pool is used for this cluster volume.
5. Click **NEXT**.
6. Review the settings for the new volume and click **CREATE** when ready.

> New cluster volumes are added to the individual Cluster cards on the TrueCommand **Clusters** screen.

Reviewer: why is cluster capitalized? if it is a general mention of a cluster don't capitalize it. If you are talking about something they see on the screen called Cluster use bold.

> New cluster volumes are added to the individual Cluster cards on the TrueCommand **Clusters** screen.

Author: ```suggestion
New cluster volumes are added to the individual cluster cards on the TrueCommand **Clusters** screen.
```

2. Choose an **ACL** to apply to the share.
   {{< expand "Current Options (click to expand)" "v" >}}
   * **POSIX_OPEN** - Template that grants read, write, and execute permissions to all users.
   * **POSIX_RESTRICTED** - Template that grants read, write, and execute to owner and group, but not other. The template may optionally include the special-purpose 'builtin_users' and 'builtin_administrators' groups as well as Domain Users and Domain Admins groups in Active Directory environments.
   {{< /expand >}}
3. Setting **Readonly** prevents users from making any changes to the cluster volume contents.
4. Click **CONFIRM** to create the SMB share and make it immediately active.

{{< hint info >}}
The SMB share is added to the SCALE **Shares** > **SMB** section for each system in the cluster.
Attempting to manage the share from the SCALE UI is not recommended.
{{< /hint >}}

### Connecting to the Shared Volume

There are many different ways to access an SMB share, but this article demonstrates using the Windows 10 File Explorer.

1. From a Windows 10 system that is connected to the same network as the clustering environment, open File Explorer.

> 2. In the Navigation bar, clear the contents and enter `\\` followed by the IP address or hostname of one of the clustered SCALE systems. Press <kbd>Enter</kbd>.

Author: ```suggestion
2. In the **Navigation** bar, clear the contents and enter `\\` followed by the IP address or host name of one of the clustered SCALE systems. Press <kbd>Enter</kbd>.
```
